packages/taquito-michelson-encoder/src/michelson-map.ts

```diff
@@ -116,7 +116,7 @@ export class MichelsonMap<K extends MichelsonMapKey, T extends any> {
     this.assertTypecheckKey(key);
 
     const strKey = this.serializeDeterministically(key);
-    return this.valueMap.get(strKey);
+    return this.valueMap.get(strKey) ? this.valueMap.get(strKey) : undefined;
```

Comment:
This can be problematic if we assume the test case:
```
it('returns the value for an existing key', () => {
  const map = new MichelsonMap();
  map.set('test', 0);
  expect(map.get('test')).toEqual(0);
});
```
Since all falsely values will default to `undefined` in this case which would be incorrect.

Reply:
ah yes, good catch @AndrewKishino !  In that case I think the code that's already implemented should do the trick. No PR needed